x-pack/filebeat/input/httpjson/request.go

```diff
@@ -163,21 +181,160 @@ func newRequester(
+				n = 0
+				continue
+			}
+			urlCopy = rf.url
+			urlString = rf.url.String()
+			// perform request over collected ids
+			for _, s := range ids {
+				// reformat urls of requestFactory using ids
+				rf.url, err = generateNewUrl(rf.replace, urlString, []byte(s))
+				if err != nil {
+					return fmt.Errorf("failed to generate new URL: %w", err)
+				}
+
+				// collect data from new urls
+				httpResp, err = rf.collectResponse(stdCtx, trCtx, r, publisher)
+				if err != nil {
+					return fmt.Errorf("failed to execute http client.Do: %w", err)
+				}
+				// store data according to response type
+				if i+1 == len(r.requestFactory) && len(ids) != 0 {
+					finalResps = append(finalResps, httpResp)
+				} else {
+					intermediateResps = append(intermediateResps, httpResp)
+				}
+			}
+			rf.url = urlCopy
+
+			if i+1 == len(r.requestFactory) {
+				split = r.responseProcessor.split
+				r.responseProcessor.split = nil
+				n, err = r.processAndPublishEvents(0, stdCtx, trCtx, publisher, finalResps)
+				if err != nil {
+					return err
+				}
+				r.responseProcessor.split = split
+				continue
+			} else {
+				ids, err = r.getIdsFromResponses(intermediateResps, r.requestFactory[i+1].replace)
+				if err != nil {
+					return err
+				}
+				split = r.responseProcessor.split
+				r.responseProcessor.split = nil
+				n, err = r.processAndPublishEvents(1, stdCtx, trCtx, publisher, intermediateResps)
+				if err != nil {
+					return err
+				}
+				r.responseProcessor.split = split
+			}
+		} else {
+			// perform and store regular call responses
+			httpResp, err = rf.collectResponse(stdCtx, trCtx, r, publisher)
+			if err != nil {
+				return fmt.Errorf("failed to execute http client.Do: %w", err)
+			}
+			if len(r.requestFactory) <= 1 {
+				finalResps = append(finalResps, httpResp)
+				n, err = r.processAndPublishEvents(0, stdCtx, trCtx, publisher, finalResps)
+				if err != nil {
+					return err
+				}
+				continue
+			} else {
+				intermediateResps = append(intermediateResps, httpResp)
+				ids, err = r.getIdsFromResponses(intermediateResps, r.requestFactory[i+1].replace)
+				if err != nil {
+					return err
+				}
+				n, err = r.processAndPublishEvents(1, stdCtx, trCtx, publisher, intermediateResps)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	httpResp.Body.Close()
+	r.log.Infof("request finished: %d events published", n)
+
+	return nil
+}
+
+// getIdsFromResponses returns ids from responses
+func (r *requester) getIdsFromResponses(intermediateResps []*http.Response, replace string) ([]string, error) {
+	var b []byte
+	var ids []string
+	var err error
+	// collect ids from all responses
+	for _, resp := range intermediateResps {
+		if resp.Body != nil {
+			b, err = ioutil.ReadAll(resp.Body)
+			if err != nil {
+				return nil, fmt.Errorf("error while reading response body: %v", err)
+			}
+		}
+		// Restore the io.ReadCloser to its original state
+		resp.Body = ioutil.NopCloser(bytes.NewBuffer(b))
```

Comment:
yeah, I think that's a good idea. what we can do is store []byte for intermediate responses and for final responses as it is. but  I think it will cause problems for pagination. as pagination is using responses to iterate.